R/utils-numeric.R

```diff
@@ -176,6 +176,104 @@ naToZero.default <- function(x) {
 }
 
 
+#' Drop columns containing all 0s
+#'
+#' This function drops all columns that have only 0 values. 
+#' @param x data.table, data.frame, list, or array
+#' @param ignoreNA boolean indicating if NA values should be ignored when determining all unique values in a column.
+#' Default is TRUE, which means a column that contains only 0 and NA values will be dropped. If FALSE, a column that contains 0s and
+#' NAs will not be dropped.
+#' @return cleanedData. A list that includes 1. "data" the result of removing some number of columns from the input x, and 2. "droppedColumns" names of the 
+#' columns that have been removed if the input x was a data.table, data.frame, or list. Otherwise "droppedColumns" contains
+#' the indicies of the input x that have been removed.
+#' @export
+dropZeroColumns <- function(x, ...) {
```

Comment:
i dont like its name. its not just dropping columns, its also returning a list of dropped columns. and its returning a different object than the one it received. at a min this should be clearer in the name of the method. 

but also, and probably ideally, wed revise our strategy? a class agnostic helper method to give you the list of columns might be all thats needed? choosing to drop them or not could the job of whoever is calling this thing?

and actually again, thinking about largely class agnostic helpers for dropping cols, have you seen purrr::discard?

Reply:
oh look at that!! Thank you purrr! `discard` will make it much easier.